### DISCUSSION TITLE
Disabling mcdev source formatting on retrieve

### DISCUSSION THREAD

**Discussion by AndrewEllis91** [Accenture/sfmc-devtools#1728] [Q&A] [answered]
Hello there!

My team started working with mcdev recently and we noticed that retrieve calls seem to be running some form of source formatting on asset files retrieved. For example, below is a snippet of AMPScript that is present within SFMC:
```
/* Logic to display Membership for Membership Types */
IF (
    @showbanner
    AND (
        (@MembershipTier == 'STUDIOMEMBER' AND @ShowMemberStudioBanner == 'false')
        OR (@MembershipTier == 'ESSENTIALMEMBER' AND @ShowEssentialBanner == 'false')
        OR (@MembershipTier == 'NONMEMBER' AND @ShowNonMemberBanner == 'false')
        OR (EMPTY(@IMAGE_1))
        OR (EMPTY(@IMAGE_LINK_1))
    )
) THEN
    SET @showbanner = false
ENDIF
```

Now, here is what is retrieved by mcdev:
```
/* Logic to display Membership for Membership Types */
    IF (         @showbanner AND (             (@MembershipTier == 'STUDIOMEMBER' AND @ShowMemberStudioBanner == 'false') OR (@MembershipTier == 'ESSENTIALMEMBER' AND @ShowEssentialBanner == 'false') OR (@MembershipTier == 'nonmember' AND @ShowNonMemberBanner == 'false') OR ( EMPTY(@IMAGE_1)) OR (EMPTY(@IMAGE_LINK_1))         )     ) THEN
        SET @showbanner = false
    ENDIF
```

Within vscode workspace settings, I have `"editor.formatOnSave": false`. Any ideas?

Thanks!

Andrew

**Comment by JoernBerkefeld** [comment]
Hm, we use beauty-amp-core2 v0.4.9 - unfortunately that has its flaws. Would it help to be able to disable auto formatting?

**Reply by AndrewEllis91** [reply]
Thanks for the quick response! It would be very helpful if we could disable this and keep the retrieved content the same format that exists within SFMC. Is that possible in the current release or would it be a new feature?

Thanks again!

**Reply by JoernBerkefeld** [reply]
I can add that in a new release. Would probably be activated via an option flag in the mcdev config

**Comment by JoernBerkefeld** [comment, marked as answer]
Created a ticket for it:
- #1729

**Reply by JoernBerkefeld** [reply]
final solution in 7.5.0:

- options.formatOnSave added to config
- retrieve & deploy command have new option --format, allowing you to override options.formatOnSave=false during runtime
- retrieve & deploy command have new option --no-format, allowing you to override options.formatOnSave=true during runtime
